pika/adapters/blocking_connection.py

```diff
@@ -700,6 +700,15 @@ def channel(self, channel_number=None):
 
         return channel
 
+    def __enter__(self):
+        # Prepare `with` context
+        return self
+
+    def __exit__(self, tp, value, traceback):
+        # Close connection after `with` context
+        self.close()
+        return isinstance(value, (KeyboardInterrupt, SystemExit, GeneratorExit))
```

Comment:
@reddec: we don't want the `return isinstance(...` statement here at all. This is because we don't want to suppress any of these exceptions in pika. This would prevent `CTL-C` from stopping the program. If user's code want's to suppress them, then user's code is free to do that on its own.

Let's have just this in `__exit__`:

```
def __exit__(self, *_args):
    self.close()
```
and add the test:
```
class TestConnectionContextManagerClosesConnectionAndPassesSystemException(BlockingTestCaseBase):
    def test(self):
        """BlockingConnection: connection context manager closes connection and passes system exception"""
        with self.assertRaises(SystemExit):
            with self._connect() as connection:
                self.assertTrue(connection.is_open)
                raise SystemExit()

        self.assertTrue(connection.is_closed)

```